packages/bsky/src/services/indexing/index.ts

```diff
@@ -111,22 +137,34 @@ export class IndexingService {
         .execute(),
     ])
     await Promise.all([
-      this.db.db.deleteFrom('assertion').where('creator', '=', did).execute(),
       this.db.db.deleteFrom('follow').where('creator', '=', did).execute(),
       this.db.db.deleteFrom('post').where('creator', '=', did).execute(),
       this.db.db.deleteFrom('profile').where('creator', '=', did).execute(),
       this.db.db.deleteFrom('repost').where('creator', '=', did).execute(),
       this.db.db.deleteFrom('vote').where('creator', '=', did).execute(),
-      this.db.db
-        .updateTable('assertion')
-        .set({
-          confirmUri: null,
-          confirmCid: null,
-          confirmCreated: null,
-          confirmIndexed: null,
-        })
-        .where('subjectDid', '=', did)
-        .execute(),
     ])
   }
 }
+
+const resolveExternalHandle = async (
+  pds: string,
+  handle: string,
+): Promise<string | undefined> => {
+  try {
+    const did = await handleLib.resolveDns(handle)
+    return did
+  } catch (err) {
+    if (err instanceof handleLib.NoHandleRecordError) {
+      // no worries it's just not found
+    } else {
+      subLogger.error({ err, handle }, 'could not resolve dns handle')
+    }
+  }
+  try {
+    const agent = new ApiAgent({ service: pds }) // @TODO all good to use pds rather than the handle itself?
+    const res = await agent.api.com.atproto.handle.resolve({ handle })
+    return res.data.did
```

Comment:
@dholms also was hoping for an extra eye on this.  Since I happen to have the user's pds from their did doc, I wondered if this may be more reliable than using their handle as the `service` without sacrificing any correctness.

Reply:
ah no unfortunately not, you should use their handle.

Reason being: I can claim the handle `dholms.xyz` _either_ using a dns record _or_ hosting a server with the handle.resolve xrpc method there

Reply:
Altho their server _could_ probably still resolve that request for you, I think it's better to go to the source